src/Index/Index.php

```diff
@@ -15,14 +15,31 @@ class Index implements ReadableIndex, \Serializable
     use EmitterTrait;
 
     /**
-     * An associative array that maps fully qualified symbol names to Definitions
+     * An associative array that maps fully qualified names to
+     *     an associative array that maps fully qualified symbol names
+     *     to global Definitions, e.g. :
+     *     [
+     *         'Psr\Log\LoggerInterface' => [
+     *             'Psr\Log\LoggerInterface->log()' => $definition,
+     *             'Psr\Log\LoggerInterface->debug()' => $definition,
+     *         ],
+     *     ]
+     *
+     * @var array
+     */
+    private $fqnDefinitions = [];
+
+    /**
+     * An associative array that maps fully qualified symbol names
+     * to global Definitions
      *
      * @var Definition[]
      */
-    private $definitions = [];
+    private $globalDefinitions = [];
```

Comment:
Why do we need two arrays?

Reply:
This is used to find the global definitions quicker than only having a single array of definitions and checking all the items one by one to get the global definitions.
See [here](https://github.com/felixfbecker/php-language-server/pull/451/files#diff-18eb38922fb7235266f15163e2b050edL319).

Reply:
Does that mean it doubles the needed memory?

Reply:
Yes, this array contains references to global definitions (which are already in the `$fqnDefinitions` array). So yes, memory grows up a little bit. I can try to have only the non global definitions in the `$fqnDefinitions`array to avoid the references duplicaiton.